capture/capture_otbn.py

```diff
@@ -0,0 +1,764 @@
+    logger.setLevel(logging.INFO)
+    console = logging.StreamHandler()
+    logger.addHandler(console)
+
+    # Parse the provided arguments.
+    args = helpers.parse_arguments(argv)
+
+    # Check the ChipWhisperer version.
+    check_version.check_cw("5.7.0")
+
+    # Load configuration from file.
+    with open(args.cfg) as f:
+        cfg = yaml.load(f, Loader=yaml.FullLoader)
+
+    # Determine the capture mode and configure the current capture.
+    mode = cfg["test"]["app"]
+    batch = cfg["test"]["batch_mode"]
+    if batch is False:
+        # For non-batch mode, make sure that num_segments = 1.
+        cfg[cfg["capture"]["scope_select"]]["num_segments"] = 1
+        logger.info(
+            "num_segments needs to be 1 in non-batch mode. Setting num_segments=1."
+        )
+
+    # Setup the target, scope and project.
+    target, scope, project = setup(cfg, args.project)
+
+    # Create capture config object.
+    capture_cfg = CaptureConfig(
+        capture_mode=mode,
+        batch_mode=batch,
+        num_traces=cfg["capture"]["num_traces"],
+        num_segments=cfg[cfg["capture"]["scope_select"]]["num_segments"],
+        output_len=cfg["target"]["output_len_bytes"],
+        text_fixed=bytearray(),
+        key_fixed=bytearray(),
+        key_len_bytes=cfg["test"]["key_len_bytes"],
+        text_len_bytes=cfg["test"]["text_len_bytes"],
+        protocol=cfg["target"]["protocol"],
+        C=bytearray(),
+        seed_fixed=bytearray(),
+        expected_fixed_key=bytearray(),
+        k_fixed=bytearray(),
+        expected_fixed_output=0)
+    logger.info(
+        f"Setting up capture {capture_cfg.capture_mode} batch={capture_cfg.batch_mode}..."
+    )
+
+    # Configure cipher.
+    ot_otbn_vert, curve_cfg = configure_cipher(cfg, target, capture_cfg)
+
+    # Capture traces.
+    if mode == "keygen":
+        capture_keygen(cfg, scope, ot_otbn_vert, capture_cfg, curve_cfg,
+                       project, target)
+    elif mode == "modinv":
+        capture_modinv(cfg, scope, ot_otbn_vert, capture_cfg, curve_cfg,
+                       project, target)
+    else:
+        # TODO: add support for modinv app
+        raise NotImplementedError('Cofigured OTBN app not yet supported.')
+
+    # Print plot.
+    print_plot(project, cfg)
+
+    # Save metadata.
+    metadata = {}
+    metadata["datetime"] = datetime.now().strftime("%m/%d/%Y, %H:%M:%S")
+    metadata["cfg"] = cfg
+    metadata["num_samples"] = scope.scope_cfg.num_samples
+    metadata["offset_samples"] = scope.scope_cfg.offset_samples
+    metadata["scope_gain"] = scope.scope_cfg.scope_gain
+    metadata["cfg_file"] = str(args.cfg)
+    metadata["fpga_bitstream"] = cfg["target"]["fpga_bitstream"]
+    # TODO: Store binary into database instead of binary path.
+    # (Issue lowrisc/ot-sca#214)
+    metadata["fw_bin"] = cfg["target"]["fw_bin"]
+    # TODO: Allow user to enter notes via CLI.
+    # (Issue lowrisc/ot-sca#213)
+    metadata["notes"] = ""
```

Comment:
```suggestion
    # Store bitstream information.
    metadata["fpga_bitstream_path"] = cfg["target"]["fpga_bitstream"]
    metadata["fpga_bitstream_crc"] = helpers.file_crc(cfg["target"]["fpga_bitstream"])
    if args.save_bitstream:
        metadata["fpga_bitstream"] = helpers.get_binary_blob(cfg["target"]["fpga_bitstream"])
    # Store binary information.
    metadata["fw_bin_path"] = cfg["target"]["fw_bin"]
    metadata["fw_bin_crc"] = helpers.file_crc(cfg["target"]["fw_bin"])
    if args.save_binary:
        metadata["fw_bin"] = helpers.get_binary_blob(cfg["target"]["fw_bin"])
    # Store user provided notes.
    metadata["notes"] = args.notes
    # Store the Git hash.
    metadata["git_hash"] = helpers.get_git_hash()
```
Could you please add the following to reflect recent changes to the metadata?

Reply:
Of course! Sorry, I apparently used an older version of capture_aes.py as a template and missed the changes when rebasing.